src/option_parser.cr

```diff
@@ -1,3 +1,30 @@
+# `OptionParser` is a class for command-line options processing. It supports:
+#
+# * Short and long modifier style options (example: `-h`, `--help`)
+# * Command style arguments (example: `crystal build`)
```

Comment:
I don't think we support that. The compiler is currently taking the first element of ARGV, then based on it creates an OptionParser that is configured depending on the command.